mods/thefirstnewgen/abilities.js

```diff
@@ -41,15 +56,33 @@ exports.BattleAbilities = {
 		name: "Burrow",
 		rating: 3,
 	},
-	"draconiclight": { /* Will code the rst of it later */
-		shortDesc: "This Pokemon's Dragon-type attacks have their power multiplied by 1.5.",
+	"draconiclight": { 
+		shortDesc: "1.5x Dragon moves and 1/2 damage from fire, water, electric, and grass, but 2x from Dragon",
 		onBasePowerPriority: 8,
 		onBasePower: function(basePower, attacker, defender, move) {
 			if (move.type === 'Dragon') {
 				this.debug('DraconicLight boost');
 				return this.chainModify(1.5);
 			}
 		},
+				onSourceModifyAtk: function (atk, attacker, defender, move) {
+			if (move.type === 'Fire' || move.type === 'Water' || move.type === 'Electric' || move.type === 'Grass') {
+				this.debug('Draconic Light weaken');
+				return this.chainModify(0.5);
+			}
+		},
+		onModifySpAPriority: 5,
```

Comment:
You sure? Read the desc and if still yes you do et cuz I won't be on PC till hmm tomorrow nite